mm2src/coins/eth/tron/address.rs

```diff
@@ -0,0 +1,151 @@
+            .into_vec()
+            .map_err(|e| format!("Invalid base58check address: {}", e))?;
+
+        if data.len() != ADDRESS_BYTES_LEN || data[0] != ADDRESS_HEX_PREFIX {
+            return Err(format!(
+                "Invalid address: expected {} bytes with prefix 0x{:x}",
+                ADDRESS_BYTES_LEN, ADDRESS_HEX_PREFIX
+            ));
+        }
+
+        let inner = data
+            .try_into()
+            .map_err(|_| "Failed to convert address bytes to array".to_string())?;
+
+        Ok(Self { inner })
+    }
+
+    /// Construct from hex string, with or without `0x` prefix.
+    pub fn from_hex(s: &str) -> Result<Self, String> {
+        let s = s.strip_prefix("0x").unwrap_or(s);
+        let data = hex::decode(s).map_err(|e| format!("Invalid hex address: {}", e))?;
+
+        if data.len() != ADDRESS_BYTES_LEN || data[0] != ADDRESS_HEX_PREFIX {
+            return Err(format!(
+                "Invalid address: expected {} bytes with prefix 0x{:x}",
+                ADDRESS_BYTES_LEN, ADDRESS_HEX_PREFIX
+            ));
+        }
+
+        let inner = data
+            .try_into()
+            .map_err(|_| "Failed to convert address bytes to array".to_string())?;
+
+        Ok(Self { inner })
+    }
+
+    /// Show as base58 string (canonical user format).
+    pub fn to_base58(&self) -> String { bs58::encode(self.inner).with_check().into_string() }
+
+    /// Show as hex string, lowercase (canonical hex format).
+    pub fn to_hex(&self) -> String { hex::encode(self.inner) }
+
+    /// Return the 21 bytes (0x41 + 20).
+    pub fn as_bytes(&self) -> &[u8] { &self.inner }
+}
+
+impl fmt::Display for Address {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { write!(f, "{}", self.to_base58()) }
+}
+
+impl fmt::Debug for Address {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Address({} / 0x{})", self.to_base58(), self.to_hex())
+    }
+}
+
+impl Serialize for Address {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_base58())
+    }
+}
+
+impl<'de> Deserialize<'de> for Address {
+    fn deserialize<D>(deserializer: D) -> Result<Address, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = <&str>::deserialize(deserializer)?;
+        Address::from_str(s).map_err(serde::de::Error::custom)
+    }
+}
+
+impl FromStr for Address {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Check for Base58 format
+        if s.len() == ADDRESS_BASE58_LEN && s.starts_with(ADDRESS_BASE58_PREFIX) {
+            return Self::from_base58(s);
+        }
+
+        // Check for hex format (with or without 0x prefix)
+        if (s.len() == ADDRESS_HEX_LEN && s.starts_with(&format!("{:x}", ADDRESS_HEX_PREFIX)))
+            || (s.len() == ADDRESS_HEX_LEN + 2 && s.starts_with(&format!("0x{:x}", ADDRESS_HEX_PREFIX)))
```

Comment:
why not just use `s.starts_with("41")` and `s.starts_with("0x41")` instead of using `format!`? formating is redundant here as I can see